core/src/main/java/org/openstack4j/openstack/barbican/domain/BarbicanSecret.java

```diff
@@ -18,16 +19,17 @@
     @JsonProperty("algorithm")
     private String algorithm;
     @JsonProperty("bit_length")
-    private int bitLength;
-    @JsonProperty("content-types")
-    private List<String> contentTypes;
+    private Integer bitLength;
+    @JsonProperty("content_types")
+    private Map<String, String> contentTypes;
     @JsonProperty("created")
-    private Date createTime;
+    private String createTime;
     @JsonProperty("updated")
-    private Date updateTime;
+    private String updateTime;
```

Comment:
Why Date -> String?

Reply:
Because it is listed as a String in the Barbican API.

See [here](https://docs.openstack.org/barbican/latest/api/reference/secrets.html#get-v1-secrets)

Reply:
I was asking, because changing the type is a breaking change and it might be more convenient for our Java client to have a Date here instead of a String. But since you're the author I would leave that up to you.
Besides this, this PR is ready to merge from my side.

Reply:
Well, i get your point & Date would probably be better. I'll change it back